### PR TITLE
Room Time Slot Matrix Added

### PR DESCRIPTION
**Pull request by wahibkapdi**
Rubocop - 25 offenses
Coverage - 
Unit - 93%
Cucumber - 89% (Missing Coverage in Time Slots)

### CODE REVIEW

File: app/controllers/room_bookings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+# Room Bookings Controller
+class RoomBookingsController < ApplicationController
+  def index
+    schedule_id = params[:schedule_id]
+    @schedule = Schedule.find(params[:schedule_id])
+    @rooms = @schedule.rooms.where(is_active: true).where.not(building_code: 'ONLINE')
+    @tabs = TimeSlot.distinct.pluck(:day)
+    @active_tab = params[:active_tab] || @tabs[0]
+    @time_slots = TimeSlot.where(time_slots: { day: @active_tab })
```
Comment (colbyendres) [inline app/controllers/room_bookings_controller.rb]: The times can appear out of order in the table. Fix this by adding:
```
@time_slots = TimeSlot.where(time_slots: {day: @active_tab}).to_a
@time_slots.sort_by!{ |ts| Time.parse(ts.start_time) }
```
which puts the times in temporal order.

**Comment by colbyendres** [comment]
Are we intending `RoomBookings#index` to be the landing page for the finished schedule or is this just for creating locked courses? If the latter, I think this table structure should be duplicated to represent the final schedule. Our scheduling algorithm could then just populate `@bookings_matrix` with a string containing course, section, and professor

**Review by colbyendres — CHANGES_REQUESTED** [review]

File: app/controllers/room_bookings_controller.rb (hunk 1 above)
Comment (wahibkapdi) [reply app/controllers/room_bookings_controller.rb]: Good catch. Fixed.

**Comment by wahibkapdi** [comment]
> Are we intending `RoomBookings#index` to be the landing page for the finished schedule or is this just for creating locked courses? If the latter, I think this table structure should be duplicated to represent the final schedule. Our scheduling algorithm could then just populate `@bookings_matrix` with a string containing course, section, and professor

We are populating RoomBookings#index with finished schedule. As generating the schedule is just going to be an intermediate step as the client keeps iterating over the schedules.

**Comment by colbyendres** [comment]
> We are populating RoomBookings#index with finished schedule. As generating the schedule is just going to be an intermediate step as the client keeps iterating over the schedules.

That seems reasonable to me. We'll have the `Generate Schedule` button on `Schedule#index` redirect to this page whenever the algorithm finishes.

**Review by colbyendres — APPROVED** [review]

**Pull request merged**